src/librustc/ty/print/pretty.rs

```diff
@@ -973,6 +973,18 @@ pub trait PrettyPrinter<'tcx>:
                 return Ok(self);
             }
         }
+
+        if let ty::FnPtr(_) = ct.ty.kind {
```

Comment:
The usage of `if let` throughout this function is pretty weird; should be rewritten with a `match` instead imo.

Reply:
I tried that before. It becomes a mess because of all the fallbacks

Reply:
In my experience that indicates that some functions need to be extracted.

Reply:
I managed to refactor it into a `match` statement, only the byte string and string case are a bit awkward.